scanpointgenerator/excluders/roiexcluder.py

```diff
@@ -36,9 +36,12 @@ def create_mask(self, x_points, y_points):
 
         mask = np.zeros_like(x_points, dtype=bool)
         for roi in self.rois:
+            x = x_points.copy()
```

Comment:
Actually - perhaps we shouldn't be doing this - maybe the regions themselves should copy the arrays if-and-only-if they need to perform calculations. It would be easier to reason about elsewhere, if we require the arrays passed are unchanged.

I only allowed it initially because CompoundGenerator was *usually* passing in new arrays to begin with (due to expanding them out).